aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -657,9 +653,18 @@ public void onSessionConnect(
         final String responseChannel,
         final byte[] encodedCredentials)
     {
-        final long clusterSessionId = Cluster.Role.LEADER == role ? nextSessionId++ : NULL_VALUE;
-        final ClusterSession session = new ClusterSession(
-            clusterSessionId, responseStreamId, refineResponseChannel(responseChannel));
+        // use null value always here.  create session id later
+        // after authentication
+        //final long clusterSessionId = Cluster.Role.LEADER == role ? nextSessionId++ : NULL_VALUE;
+        /**
+         * let's try a new strategy here.  set the session id here and increment the counter
+         * if the session is rejected, make this session id available again.  is it possible
+         * for several sessions to be in process of authentication at a time?  if so, then this
+         * gets more complicated.
+         */
+        final long sessionId = getNewSessionId();
```

Comment:
This approach is much more complicated and error prone than the original one with not assigning an id until session is authenticated. Why was this approaches chosen?